share/wake/lib/core/result.wake

```diff
@@ -95,18 +102,21 @@ export def rmap (fn: a => b): Result a fail => Result b fail = match _
     Pass a = Pass (fn a)
     Fail f = Fail f
 
+
 # rmapPass: apply a fallible function a Pass-ing result
 # If you find yourself using the function, consider using require instead.
 export def rmapPass (fn: a => Result b fail): Result a fail => Result b fail = match _
     Pass a = fn a
     Fail f = Fail f
 
+
 # Applies a fallible function to Fail value or propogates Pass
 # If you find yourself using the function, consider using require instead.
 export def rmapFail (fn: a => Result pass b): Result pass a => Result pass b = match _
     Pass a = Pass a
     Fail f = fn f
```

Comment:
Although I have been trying to abide by this convention since a lot of Wake code uses it and I know this has probably been discussed a lot before, it would be my personal preference to put `match` on a newline. (I think this is what @mmjconolly comment also alluded to, but I could be mistaken.)

```suggestion
export def rmapFail (fn: a => Result pass b): Result pass a => Result pass b =
    match _
        Pass a = Pass a
        Fail f = fn f
```

If `match` didn't require the additional indentation, even better. :)

Aside from personal preference, I feel like in a lot of these cases, I don't notice the `match` as it just kinda hangs out over there ---->. It also breaks some consistency of my preferred formatting of top-level functions:

```wake
def <functionName> <args> : <returnType> =
    <expression>
```